### PR TITLE
Refresh HTML manual/API docs style

### PR DESCRIPTION
**Pull request by yawaramin**
Simplify colours and get rid of gradients. Also fix search button icon.

Before:

Screenshot 2023-12-29 at 11 00 37

After:

Screenshot 2023-12-29 at 01 01 14

### CODE REVIEW

**Comment by sidkshatriya** [comment]
Suggestion: please attach a _before_ screenshot for easy comparison. So it can be compared visually with the _after_ screenshot you posted. This way we don't need to actually go to the site to compare with this.

Also screenshot to screenshot comparison is easier since the size, dpi, color temperature etc. would be the same in both.

**Comment by yawaramin** [comment]
After further reducing the top border of each item:

Screenshot 2023-12-29 at 11 02 18

**Comment by wikku** [comment]
That's a big improvement IMO. 

The top border clearly delineates items and communicates that I should be reading the description below and not above, currently only indicated by slightly more whitespace below descriptions. Though I'd use `border-top: 2px`. 

The screenshot still shows odd alignment of the description and "Raises" warnings, I think aligning them both with `margin-left: 1.5 rem` would look good.

image
(This was just playing with Firefox dev tools)

**Comment by yawaramin** [comment]
Thanks! Adjusted description items left margin:

Screenshot 2023-12-29 at 12 22 22

**Review by SGrondin — CHANGES_REQUESTED** [review]
After this change there's no need to download the `search_icon.svg` anymore, so it should probably be removed from the Makefile: [here](https://github.com/ocaml/ocaml/blob/d8e2f5f0c069370b78cfb41b0d730360f07b1d74/manual/src/html_processing/Makefile#L55-L57) and [here](https://github.com/ocaml/ocaml/blob/d8e2f5f0c069370b78cfb41b0d730360f07b1d74/manual/src/html_processing/Makefile#L76)

Note: I agree, I think it's a huge improvement

**Comment by yawaramin** [comment]
Awesome, thanks! Removed from Makefile, and no wonder I couldn't find that file in the repo.

**Review by SGrondin — APPROVED** [review]
With the Makefile change it looks good to me

**Comment by yawaramin** [comment]
Looks like the manual and API docs styles are entangled. Need to fix this first.

Screenshot 2023-12-30 at 13 05 40

**Comment by yawaramin** [comment]
Manual:

Screenshot 2023-12-30 at 15 22 26

API docs:

Screenshot 2023-12-30 at 15 22 51

**Comment by gasche** [comment]
For code blocks in the manual I don't understand why you are adding a top border. In the doc it makes sense because the item is always related to the doc that follows, not to the content before, so you add a visual separator above. But this is not true for arbitrary code blocks in the manual, including in the examples in your screenshot -- `fib` is related to the text right above it.

**Comment by yawaramin** [comment]
I guess for consistency with the API docs, but you have a point that the delineation separates the prose from its related code. How about a bottom border instead? I feel like some structure in the code formatting is helpful.

Screenshot 2024-01-02 at 16 44 45

**Comment by Octachron** [comment]
Thanks for the changes. The changes look good to me for my non-expert eyes. 

I think we want to avoid visible horizontal lines as delimiters on the manual side, both top and bottom since the code
is often part of the text there. Maybe adding a vertical border on the left or right side could work better? 

Looking at ocaml.org for another inspiration, the tutorial are using round border (but with syntactic color highlighting).

**Comment by yawaramin** [comment]
Thanks @Octachron for the review. I'm not terribly attached to the borders so just removed them for the manual. Now it looks like this:

Screenshot 2024-01-03 at 19 07 26

File: manual/src/html_processing/scss/_common.scss [hunk 1]
```diff
@@ -143,7 +143,9 @@ html {
 		display:block;
 		&:hover {
 		    box-shadow:none;
-		    background-color: #edbf84;}
+		    background-color: white;
+        text-decoration: underline;
+        }
```
Comment (gasche) [inline manual/src/html_processing/scss/_common.scss]: nit: the identation looks off here

File: manual/src/html_processing/scss/manual.scss [hunk 2]
```diff
@@ -247,7 +246,7 @@ nav.toc{
 }
 
 pre{
-    background:linear-gradient(to left,#fff 0,#ede8e5 100%)
+	background-color: #ede8e5;
```
Comment (gasche) [inline manual/src/html_processing/scss/manual.scss]: nit: indentation

File: manual/src/html_processing/scss/style.scss [hunk 3]
```diff
@@ -199,7 +198,7 @@
 	/*box-shadow: 0 0px 0 1px rgba(255, 215, 181, 0.8) !important;*/
 	border-radius: 1px;
 	/*border-bottom: 4px solid rgb(255, 215, 181);*/
-	box-shadow: 0 4px 0 0px rgb(255, 215, 181);
+  background-color:#ffd6b5;
```
Comment (gasche) [inline manual/src/html_processing/scss/style.scss]: nit: indentation

**Review by gasche — APPROVED** [review]
The result looks good to me now. Approved.

There are a few places in the code where the PR appears to break indentation, could you fix them?

Can you clean the history to avoid the back-and-forth?

**Comment by yawaramin** [comment]
@gasche sure, did the requested clean-ups.

**Comment by gasche** [comment]
@Octachron do you want to have another round of feedback? Otherwise please feel free to merge.

**Comment by gasche** [comment]
(It would be nice to have the changes in the 5.2 version of the manual)

File: manual/src/html_processing/src/process_api.ml [hunk 4]
```diff
@@ -25,12 +25,12 @@ let search_widget with_description =
...
     else "" in
-  sprintf {|<div class="api_search"><input type="text" name="apisearch" id="api_search" class="api_search"
+  sprintf {|<div class="api_search"><input type="search" name="apisearch" id="api_search" class="api_search"
 	 oninput    = "mySearch(%b);"
          onkeypress = "this.oninput();"
          onclick    = "this.oninput();"
 	 onpaste    = "this.oninput();">
-<img src="search_icon.svg" alt="Search" class="api_search svg" onclick="mySearch(%b)">%s</div>
+<button title="Search" onclick="mySearch(%b)" style="cursor:pointer">🔎</button> %s</div>
```
Comment (Octachron) [inline manual/src/html_processing/src/process_api.ml]: This raises a small concern in term of accessibility. I wonder if replacing 🔎 with
```html
<span aria-hidden="true">🔎</span>Search
```
would not be better?

Comment (yawaramin) [reply manual/src/html_processing/src/process_api.ml]: Thanks for the suggestion, I've just done this. Looks like:

Screenshot 2024-01-04 at 13 37 14

**Comment by gasche** [comment]
Merged, thanks!

**Pull request merged**